### PR TITLE
update readme - clarification on debian server setup

### PR DESCRIPTION
**Pull request by MartinHenriksen**
this PR addresses #123 - very small changes to the readme

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -56,26 +56,28 @@ Note: You're not restricted to digital ocean. Any machine with docker should be
 
        $ git clone https://github.com/speckleworks/SpeckleServer.git
        $ cd SpeckleServer
-       $ npm install
+       $ sudo npm install
```
Comment (didimitrie) [inline readme.md]: Theoretically, doing `sudo` on `npm install` is bad, especially in production. Was it really needed?

```suggestion
       $ npm install
```

We could add "when getting permission denied errors, preface your commands with `sudo` as generic help item.  [Obligatory reference](https://xkcd.com/149/)

File: readme.md [hunk 2]
```diff
@@ -56,26 +56,28 @@ Note: You're not restricted to digital ocean. Any machine with docker should be
...
 
-5) Follow the instructions in `.env-base` file to configure your server.
+5) Follow the instructions in `.env-base` file to configure your server. Use `sudo nano .env` to edit.
```
Comment (didimitrie) [inline readme.md]: again, theoretically, if you're logged in as a non-sudo user, you should be able to edit the files from your home dir, where theoretically you've cloned the repo...

File: readme.md [hunk 3]
```diff
@@ -56,26 +56,28 @@ Note: You're not restricted to digital ocean. Any machine with docker should be
...
 
-       $ mongo
-       $ redis-cli
+       $ service mongodb status
+       $ redis-cli ping
     
 9) Start Speckle in a third terminal: 
 
-       $ node server.js
+       $ sudo node server.js
```
Comment (didimitrie) [inline readme.md]: again, that sudo is risky! would also add instructions or a pointer towards either pm2 [0] or forever [1] as a way to keep the server alive throughout restarts/errors.
[0] http://pm2.keymetrics.io/
[1] https://www.npmjs.com/package/forever

**Comment by didimitrie** [comment]
Hello Mr. @MartinHenriksen! Good to see you around these parts :) I've added some comments to the pr, let's see how we go about them!

File: readme.md (hunk 1 above)
Comment (MartinHenriksen) [reply readme.md]: Did not know sudo was bad in this case. It was needed in my case.

File: readme.md (hunk 2 above)
Comment (MartinHenriksen) [reply readme.md]: as a non-admin user I got 'access restricted' when trying to save canges - was only possible with adding sudo to the command

**Comment by MartinHenriksen** [comment]
It looks like the the permissions of the SpeckleServer-folder is the reason sudo was required:
image
Would it be safe to give write access to the folder for non-root users? @didimitrie

**Comment by logan12358** [comment]
It looks like you're trying to install things from a folder you don't have permissions for. These steps just worked on a fresh Debian 9 install:

As root:
1. `curl -sL https://deb.nodesource.com/setup_10.x | bash` to get packages from NodeSource
2. `apt install mongodb redis-server nodejs git` to install things
3. `adduser speckle` to create a new user
4. `su - speckle` to become the speckle user

As speckle:

5. `git clone https://github.com/speckleworks/SpeckleServer.git` to get the SpeckleServer source
6. `cd ~/SpeckleServer`
7. `npm install` to get dependencies
8. `cp .env-base .env` to get base configuration. Edit `.env` at this point.
9. `cd ~/SpeckleServer/plugins`
10. `git clone https://github.com/speckleworks/SpeckleAdmin` to get the admin interface
11. `git clone https://github.com/speckleworks/SpeckleViewer` to get the 3D viewer
12. `cd ~/SpeckleServer`
13. `npm start` to start the server on port 3000

**Comment by logan12358** [comment]
Note that those steps don't produce a "proper" deployment. You probably want to:
* Reverse-proxy to port 80 through nginx or similar
* Set up a firewall
* Run the server process using [one of the tools Dimitrie mentioned](https://github.com/speckleworks/SpeckleServer/pull/124#discussion_r263521229), or by setting up a systemd unit

**Comment by MartinHenriksen** [comment]
@logan12358 you are right it does look like it is a problem with permissions, but I don't know how it should be resolved in a way that would work for all.
I setup a Ubuntu instance on ec2 and in that case I would still need sudo to follow the steps you describe; even the `adduser` only works as `sudo adduser`.
Maybe there is some note on deploying the instance that could help avoid the need for sudo

**Comment by logan12358** [comment]
The first steps (under "As root") should be performed with root permissions, either by logging in as root or by using sudo. The steps after that can be performed as the new speckle user without permissions.

The steps performed with root permissions are unavoidable on Debian or Ubuntu, as they involve system-wide installations.

**Comment by MartinHenriksen** [comment]
updated instructions with inputs from @logan12358 and @didimitrie

**Comment by MartinHenriksen** [comment]
closing this as it is superseded by #126

**Pull request closed without merging**